### PR TITLE
e2e

### PR DESCRIPTION
**Pull request by narkarum**

### CODE REVIEW

File: e2e/vpc/constants.go [hunk 1]
```diff
@@ -11,13 +11,14 @@
...
-	generation        = "gt"
-	vpcProfile        = "general-purpose"
-	vpcConfigFilePath = "/src/github.com/IBM/ibmcloud-storage-volume-lib/e2e/config/vpc-config.toml"
+	vpcZone                = "VPC_ZONE"
+	resourceGroupID        = "RESOURCEGROUP"
+	volumeSize             = 10
+	iops                   = "0"
+	volumeName             = "e2e-storage-volume"
+	volumeType             = "vpc-block"
+	generation             = "gt"
```
Comment (arahamad) [inline e2e/vpc/constants.go]: I think gt is not supported yet, are we using this value?

File: e2e/vpc/constants.go [hunk 2]
```diff
@@ -11,13 +11,14 @@
...
+	vpcProfile             = "general-purpose"
+	vpcConfigFilePath      = "/src/github.com/IBM/ibmcloud-storage-volume-lib/e2e/config/vpc-config.toml"
```
Comment (arahamad) [inline e2e/vpc/constants.go]: Please check in this file as well for a reference

File: e2e/vpc/constants.go [hunk 3]
```diff
@@ -11,13 +11,14 @@
...
+	volumeEncryptionKeyCRN = "ENCRYPTION_KEY_CRN"
```
Comment (arahamad) [inline e2e/vpc/constants.go]: This needs to be exported before executing e2e ? same for vpcZone and resourceGroupID

File: e2e/vpc/vpc_create_volume_custom_profile.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+		Iops := iops
+
+		volume := &provider.Volume{}
+
+		volume.VolumeType = volumeType
+		volume.VPCVolume.Generation = generation
+		volume.VPCVolume.ResourceGroup = &provider.ResourceGroup{}
+		volume.VPCVolume.Profile = &provider.Profile{Name: "custom"}
+		volume.Name = &volName
+		volume.Capacity = &volSize
```
Comment (arahamad) [inline e2e/vpc/vpc_create_volume_custom_profile.go]: Can we verify custom class with 2-3 different volume size as well

File: e2e/vpc/vpc_create_volume_10iops_tier.go [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+	"go.uber.org/zap"
+
+	"github.com/IBM/ibmcloud-storage-volume-lib/lib/provider"
+)
+
+var _ = Describe("ibmcloud-storage-volume-lib", func() {
+	It("VPC: Create and delete VPC volume[profile: 10iops-tier]", func() {
+		volName := volumeName + "10iops"
+		volSize := volumeSize
+		Iops := iops
```
Comment (arahamad) [inline e2e/vpc/vpc_create_volume_10iops_tier.go]: Can we avoid setting Iops for `tiered` profiles as that does not require to set the iops, applicable for other profiles as well

File: e2e/vpc/vpc_create_volume_10iops_tier.go [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+		volume.VPCVolume.Generation = generation
+		volume.VPCVolume.ResourceGroup = &provider.ResourceGroup{}
+		volume.VPCVolume.Profile = &provider.Profile{Name: "10iops-tier"}
+		volume.Name = &volName
+		volume.Capacity = &volSize
+		volume.Iops = &Iops
+		volume.VPCVolume.ResourceGroup.ID = resourceGroupID
+		volume.Az = vpcZone
+
+		volume.VPCVolume.Tags = []string{"Testing VPC Volume"}
```
Comment (arahamad) [inline e2e/vpc/vpc_create_volume_10iops_tier.go]: Can we have this `"Testing VPC Volume"` string as const and use for other volume creation and please change this string to something like "Testing VPC Volume from IBM common lib"

**Comment by arahamad** [comment]
@narkarum , Can you please share steps how to execute these e2e tests

File: e2e/vpc/constants.go (hunk 3 above)
Comment (narkarum) [reply e2e/vpc/constants.go]: We are getting it from secret. Details are in .travis.yaml

File: e2e/vpc/vpc_create_volume_custom_profile.go (hunk 4 above)
Comment (narkarum) [reply e2e/vpc/vpc_create_volume_custom_profile.go]: Sure

File: e2e/vpc/vpc_create_volume_10iops_tier.go (hunk 5 above)
Comment (narkarum) [reply e2e/vpc/vpc_create_volume_10iops_tier.go]: It profiles name.

```
    {
       "first":{
          "href":"https://us-south.iaas.cloud.ibm.com/v1/volume/profiles?limit=50"
       },
       "limit":50,
       "total_count":4,
       "profiles":[
          {
             "name":"10iops-tier",
             "family":"tiered",
             "generation":"gc",
             "href":"https://us-south.iaas.cloud.ibm.com/v1/volume/profiles/10iops-tier"
          },
          {
             "name":"5iops-tier",
             "family":"tiered",
             "generation":"gc",
             "href":"https://us-south.iaas.cloud.ibm.com/v1/volume/profiles/5iops-tier"
          },
          {
             "name":"custom",
             "family":"custom",
             "generation":"gc",
             "href":"https://us-south.iaas.cloud.ibm.com/v1/volume/profiles/custom"
          },
          {
             "name":"general-purpose",
             "family":"tiered",
             "generation":"gc",
             "href":"https://us-south.iaas.cloud.ibm.com/v1/volume/profiles/general-purpose"
          }
       ]
    }
```

File: e2e/vpc/vpc_create_volume_10iops_tier.go (hunk 6 above)
Comment (narkarum) [reply e2e/vpc/vpc_create_volume_10iops_tier.go]: I am customizing it.

File: e2e/vpc/constants.go (hunk 1 above)
Comment (narkarum) [reply e2e/vpc/constants.go]: Currently, vpc backend is not using this flag but we are setting.

**Pull request merged**